CMSgov/eAPD
# Tforkner/4030 subform cancel

Resolves #4030 

### Description
There was an issue where updates were being saved in metrics even when the user clicked cancel. I have updated metrics and the other subforms, to not save anything that was updated if the user clicks cancel.

### Automated test cases written
added a cypress test to add info to a metric and then click cancel

### Steps to manually verify this change

1. Create an outcome with a metric
2. edit the metric and click cancel
3. verify the changes you made were not saved

Please test by trying difference cases of canceling for all subforms: 
Key State Personnel
Outcomes and Metrics
Milestones
State Staff
State Expenses
Private Contractor Costs

### This pull request is ready to review when


### This pull request can be merged when
